src/client/auth/mod.rs

```diff
@@ -165,6 +189,15 @@ impl FromStr for AuthMechanism {
             MONGODB_X509_STR => Ok(AuthMechanism::MongoDbX509),
             GSSAPI_STR => Ok(AuthMechanism::Gssapi),
             PLAIN_STR => Ok(AuthMechanism::Plain),
+
+            #[cfg(feature = "tokio-runtime")]
+            MONGODB_AWS_STR => Ok(AuthMechanism::MongoDbAws),
+            #[cfg(not(feature = "tokio-runtime"))]
+            MONGODB_AWS_STR => Err(ErrorKind::ArgumentError {
```

Comment:
I figured a custom error message for this case would mitigate potential confusion from users in the case that they try to use AWS auth with async-std.